components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Testing/src/Tests/HistoricalAccess/Extensions/VariantEncoderEx.cs

```diff
@@ -0,0 +1,105 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.OpcUa.Protocol {
+    using Microsoft.Azure.IIoT.OpcUa.Twin.Models;
+    using Microsoft.Azure.IIoT.Serializers;
+    using Opc.Ua;
+    using System;
+    using System.Linq;
+
+    /// <summary>
+    /// Variant encoder extensions to encode and decode details and results
+    /// </summary>
+    internal static class VariantEncoderEx {
```

Comment:
Please use variant encoder from OPC Foundation stack.

Reply:
No, not in this change, too risky and large change.  Please handle through a separate work item.